pyvrp/cpp/search/Route.h

```diff
@@ -53,11 +50,14 @@ class Route
     ProblemData const &data;
     size_t const vehicleType_;
 
-    std::vector<Node *> nodes;  // List of nodes in this route, excl. depot
-    std::pair<double, double> centroid;  // Center point of route's clients.
+    std::vector<Node *> nodes;      // Nodes in this route, excl. depot
+    std::vector<Load> cumLoad;      // Cumulative load along route (incl.)
+    std::vector<Distance> cumDist;  // Cumulative dist along route (incl.)
```

Comment:
My main motivation for now is to move all this over to Route. We can later on rethink this a bit, once it's all nicely encapsulated inside the `Route` object. I suspect structs might be a little less cache friendly, but I doubt that matters here. Caches are large and the structs small - and they're in contiguous memory in either case. So then it's probably mostly preference.

I haven't really figured out the long term data layout of the `Route` object, so this will probably change a bit again.